joblib/test/test_hashing.py

```diff
@@ -83,28 +76,26 @@ def f(self, x):
 ###############################################################################
 # Tests
 
-def test_trival_hash():
-    """ Smoke test hash on various types.
-    """
-    obj_list = [1, 2, 1., 2., 1 + 1j, 2. + 1j,
-                'a', 'b',
-                (1, ), (1, 1, ), [1, ], [1, 1, ],
-                {1: 1}, {1: 2}, {2: 1},
-                None,
-                gc.collect,
-                [1, ].append,
-                # Next 2 sets have unorderable elements in python 3.
-                set(('a', 1)),
-                set(('a', 1, ('a', 1))),
-                # Next 2 dicts have unorderable type of keys in python 3.
-                {'a': 1, 1: 2},
-                {'a': 1, 1: 2, 'd': {'a': 1}},
-                ]
-    for obj1 in obj_list:
-        for obj2 in obj_list:
-            # Check that 2 objects have the same hash only if they are
-            # the same.
-            yield assert_equal, hash(obj1) == hash(obj2), obj1 is obj2
+@parametrize('obj1, obj2', list(itertools.product(
```

Comment:
I thought of doing this, but then I stopped, thinking that for the first look I should not demonstrate my diff which introduces a _not so special_ list in global scope.

But now we're on the same boat, so modifying it...